src/adapter/http/src/middleware/headers.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use axum::headers::authorization::Bearer;
+use axum::headers::{Authorization, Error, Header};
+use axum::TypedHeader;
+use http::{HeaderName, HeaderValue};
+
+pub type BearerHeader = TypedHeader<Authorization<Bearer>>;
+pub type VersionHeaderTyped = TypedHeader<VersionHeader>;
+
+#[derive(Debug)]
+pub struct VersionHeader(pub i32);
+
+impl Header for VersionHeader {
+    fn name() -> &'static HeaderName {
+        static NAME: HeaderName = HeaderName::from_static("x-websocket-version");
```

Comment:
Very nice use of typed headers!

`x-websocket-version` sounds like we're versioning the WS protocol itself (which has its own `Sec-WebSocket-Version` header)

maybe let's name it `x-odf-smtp-version`?

Reply:
And mb change the type name to `OdfSmtpVersion` to make it clear what we're versioning